pkg/stanza/fileconsumer/attrs/attrs.go

```diff
@@ -32,6 +39,28 @@ func (r *Resolver) Resolve(path string) (attributes map[string]any, err error) {
 	if r.IncludeFilePath {
 		attributes[LogFilePath] = path
 	}
+	if r.IncludeFileOwner || r.IncludeFileGroup {
+		var file, fileErr = os.OpenFile(fmt.Sprint(path), os.O_RDONLY, 0000)
+		if fileErr == nil {
+			var fileInfo, errStat = file.Stat()
+			if errStat == nil {
+				var fileStat = fileInfo.Sys().(*syscall.Stat_t)
+				if r.IncludeFileOwner {
+					var fileOwner, errFileUser = user.LookupId(fmt.Sprint(fileStat.Uid))
+					if errFileUser == nil {
+						attributes[LogFileOwner] = fileOwner.Username
+					}
+				}
+				if r.IncludeFileGroup {
+					var fileGroup, errFileGroup = user.LookupGroupId(fmt.Sprint(fileStat.Gid))
+					if errFileGroup == nil {
+						attributes[LogFileGroup] = fileGroup.Name
+					}
+				}
+			}
+			defer file.Close()
+		}
+	}
 	if !r.IncludeFileNameResolved && !r.IncludeFilePathResolved {
```

Comment:
This is a lot of deep nesting and to me indicates that we should perform all this logic in a dedicated function. This would allow us to return early from that function when there are simple cases.